modules/dmrpp_module/DmrppArray.cc

```diff
@@ -2965,12 +3021,14 @@ void DmrppArray::read_array_of_structure(vector<char> &values) {
 
     size_t values_offset = 0;
     int64_t nelms = this->length_ll();
+if (this->twiddle_bytes()) 
+    BESDEBUG(dmrpp_3, prolog << "swap bytes " << endl);
 
     for (int64_t element = 0; element < nelms; ++element) {
 
         auto dmrpp_s = dynamic_cast<DmrppStructure*>(var()->ptr_duplicate());
```

Comment:
I wonder if we should be checking if dmrpp_s is null here?

Reply:
Yes, add it in the new pull request.